easybuild/framework/easyconfig/format/yeb.py

```diff
@@ -94,6 +95,9 @@ def parse(self, txt):
         """
         txt = self._inject_constants_dict(txt)
         self.parsed_yeb = yaml.load(txt)
+        # parse dependencies upfront; this cant be done with simple type checking
```

Comment:
`s/cant/can't/` (or "can not")

also mention why it can not be done with simple type checking?